players.js

```diff
@@ -259,9 +259,46 @@ class Player {
 			this.proxy.PreviousRemote()
 	}
 	activatePlayer(){
-		if(this.desktopApp){
-			let app = Shell.AppSystem.get_default().lookup_app(this.desktopApp);
-			app.activate();
+		let focusedWindow = global.display.get_focus_window();
+
+		if (!this.playerWindow)
+			this.playerWindow = this._guessAppWindow(this.identity);
+
+		if (!this.playerWindow)
+			return
```

Comment:
I'm not convinced this is necessary as we can still activate the app based on the .desktop, especially using the latest implementation